trie/BaseNode.py

```diff
@@ -0,0 +1,194 @@
+from trie.constants import (
+    NODE_TYPE_BRANCH,
+    NODE_TYPE_EXTENSION,
+    NODE_TYPE_LEAF,
+)
+from trie.utils_refactor import (
+    get_common_prefix,
+)
+
+
+# NOTE: HERE ALL THE KEYS ARE NIBBLES(ints b/w 0 and 15) AND NOT STRINGS
+
+
+class BranchNode:
+    node_type = NODE_TYPE_BRANCH
+    # Keys are ints from 0 to 15, values are the child nodes
+    child_nodes = {}
```

Comment:
FYI, this results in all instances of `BranchNode` sharing `child_nodes`.  The actual implementation of `__init__` prevents this bug from being exposed but this still needs to be fixed.  Basically, classes should **never** have mutable data structures as class properties.  Changing this to be `None` is adequate to fix this.